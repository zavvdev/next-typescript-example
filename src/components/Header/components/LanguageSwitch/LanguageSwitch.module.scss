@import "src/styles/theme";

.root {
  display: flex;
  align-items: center;

  .langWrap {
    & span {
      cursor: pointer;
    }

    .lang {
      text-transform: uppercase;
      padding: 10px;
      text-decoration: none;
      color: $cBlack;
      font-size: 13px;

      &.active {
        color: $cPrimary;
        font-weight: bold;
      }
    }
  }
}