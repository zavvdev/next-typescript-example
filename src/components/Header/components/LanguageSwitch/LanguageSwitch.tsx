import cn from 'classnames';
import { useRouter } from 'next/router';
import Link from 'next/link';
import { APP_LANGUAGES } from 'src/constants';
import classes from 'src/components/Header/components/LanguageSwitch/LanguageSwitch.module.scss';

const LanguageSwitch = () => {
  const languages = Object.values(APP_LANGUAGES);
  const { locale, route } = useRouter();

  return (
    <div className={classes.root}>
      {languages.map((lang, index) => (
        <div className={classes.langWrap} key={lang}>
          <Link
            href={route}
            locale={lang}
          >
            <a className={cn(classes.lang, {
              [classes.active]: locale === lang,
            })}
            >
              {lang}
            </a>
          </Link>
          {index === 0 ? <span>|</span> : null}
        </div>
      ))}
    </div>
  );
};

export default LanguageSwitch;
