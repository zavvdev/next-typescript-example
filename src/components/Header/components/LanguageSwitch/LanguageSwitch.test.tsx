import render from 'src/tests/utils/render';
import LanguageSwitch from 'src/components/Header/components/LanguageSwitch/LanguageSwitch';

describe('<LanguageSwitch />', () => {
  test('should render LanguageSwitch component (snapshot)', () => {
    const { asFragment } = render(<LanguageSwitch />);
    expect(asFragment()).toMatchSnapshot();
  });
});
