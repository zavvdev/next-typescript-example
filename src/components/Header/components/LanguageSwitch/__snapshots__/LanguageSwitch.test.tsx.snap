// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSwitch /> should render LanguageSwitch component (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="langWrap"
    >
      <a
        class="lang"
        href="/"
      >
        en
      </a>
      <span>
        |
      </span>
    </div>
    <div
      class="langWrap"
    >
      <a
        class="lang"
        href="/"
      >
        ru
      </a>
    </div>
  </div>
</DocumentFragment>
`;
